tests/rptest/tests/scram_test.py

```diff
@@ -426,7 +426,7 @@ def __init__(self, test_context):
     def generate_string_with_control_character(length: int):
         rv = ''.join(
             random.choices(string.ascii_letters + CONTROL_CHARS, k=length))
-        if not any(char in rv for char in CONTROL_CHARS):
+        while not any(char in rv for char in CONTROL_CHARS):
```

Comment:
nit: I think it would be easier to read if the previous line was: `rv = ''`

Reply:
I also agree that expanding lines 427-429 would make the code easier to read.